openupgrade_scripts/scripts/hr/16.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Coop IT Easy (https://coopiteasy.be)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+_moved_fields = [
+    "__last_update",
+    "_order",
+    "display_name",
+    "name",
+]
+
+_xmlid_renames = [
+    (
+        "hr_contract.access_hr_contract_type_manager",
+        "hr.access_hr_contract_type_manager",
+    ),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.update_module_moved_models(
```

Comment:
Do you know what's the behavior of the method if the script is run on a DB where `hr` module was installed, but not `hr_contract`?

Reply:
Yes, in that case, the function has no effect, because the model `hr.contract.type` does not exist, neither the module `hr_contract` in the tables: `ir_model_data`, `ir_model_constraint` and `ir_model_relation`.

Therefore, the model `hr.contract.type` will be created during the update of the `hr` module, after the pre-migration script.

Reply:
Here is the details of a run on a database with only `hr` installed:
```
2023-06-21 08:50:50,691 1 INFO odooce_hr OpenUpgrade: Moving model hr.contract.type from module 'hr_contract' to module 'hr' 
2023-06-21 08:50:50,691 1 DEBUG odooce_hr OpenUpgrade: 0 rows affected after 0:00:00.000329 running UPDATE ir_model_data SET module='hr' WHERE model = 'hr.contract.type' AND module = 'hr_contract' 
2023-06-21 08:50:50,692 1 DEBUG odooce_hr OpenUpgrade: 0 rows affected after 0:00:00.000269 running UPDATE ir_model_data SET module='hr' WHERE name='model_' || 'hr_contract_type' AND model = 'ir.model' AND module = 'hr_contract' RETURNING 
res_id 
2023-06-21 08:50:50,692 1 DEBUG odooce_hr OpenUpgrade: 0 rows affected after 0:00:00.000396 running UPDATE ir_model_data imd
        SET module = 'hr'
        FROM ir_model_fields imf
        WHERE imd.model = 'ir.model.fields'
            AND imd.name = 'field_' || 'hr_contract_type' || '__' || imf.name
            AND imf.model = 'hr.contract.type' AND imd.module = 'hr_contract' 
2023-06-21 08:50:50,692 1 INFO odooce_hr OpenUpgrade: Moving fields __last_update, _order, display_name, name in model hr.contract.type from module 'hr_contract' to module 'hr' 
2023-06-21 08:50:50,693 1 DEBUG odooce_hr OpenUpgrade: 0 rows affected after 0:00:00.000486 running 
        UPDATE ir_model_data imd
        SET module = 'hr'
        FROM ir_model_fields imf
        WHERE
            imf.model = 'hr.contract.type' AND
            imf.name IN ('__last_update', '_order', 'display_name', 'name') AND
            imd.module = 'hr_contract' AND
            imd.model = 'ir.model.fields' AND
            imd.res_id = imf.id AND
            imd.id NOT IN (
               SELECT id FROM ir_model_data WHERE module = 'hr'
            )
         
2023-06-21 08:50:50,693 1 INFO odooce_hr OpenUpgrade: XMLID hr_contract.access_hr_contract_type_manager not found when renaming to hr.access_hr_contract_type_manager 
```